### PR TITLE
Fix typo ,format source code and replace `set([...])` call with set literal `{...}`

### PR DESCRIPTION
**Pull request by imbillow**

### CODE REVIEW

**Comment by XVilka** [comment]
There are some actions that still failing.

**Comment by imbillow** [comment]
> 
> 
> There are some actions that still failing.

I try to fix them

**Comment by imbillow** [comment]
I cannot reproduce the failed test on my own PC, I'm using Windows 10 1909 and Python 3.6.8.

`FAILED tests/test_issue22.py::test_issue22-MemoryError`

```
> py.test .\tests\test_issue22.py
================================================= test session starts =================================================
platform win32 -- Python 3.6.8, pytest-5.4.3, py-1.8.2, pluggy-0.13.1
rootdir: ...
plugins: cov-2.10.0
collected 1 item

tests\test_issue22.py .                                                                                          [100%]

================================================== 1 passed in 2.51s ==================================================
```

**Comment by imbillow** [comment]
https://github.com/williballenthin/python-idb/pull/76#issuecomment-649452084

**But when I run the complete test**, 

```
py.test ./tests/ -v --cov=idb
```

using Ubuntu20.04+Python2.7.10, it crashed.

image

**Comment by XVilka** [comment]
Maybe try with the profiler? I think it might be a problem of the "zip" module.

**Comment by imbillow** [comment]
> Maybe try with the profiler? I think it might be a problem of the "zip" module.

Maybe you mean `zlib` ?

I see this  `zlib.decompress()` allocate ~560MB memory.

```
Filename: .../python-idb/idb/fileformat.py

Line #    Mem usage    Increment   Line Contents
================================================
   118     47.4 MiB     47.4 MiB       @profile()
   119                                 def pcb__contents(self):
   120     47.4 MiB      0.0 MiB           if not self.header.is_compressed:
   121                                         self.contents = self._contents
   122                                     else:
   123    559.5 MiB    512.1 MiB               self.contents = zlib.decompress(self._contents)
   124    559.5 MiB      0.0 MiB               logger.debug('decompressed parsed section.')

```

This `db.vsParse()` also allocate lots of memory, it inherited from v `vstruct` class.

```
Filename: /home/iov/Source/imbillow/python-idb/idb/__init__.py

Line #    Mem usage    Increment   Line Contents
================================================
    34     46.5 MiB     46.5 MiB   @profile
    35                             def from_buffer(buf):
    36                                 # break import cycle
    37     46.7 MiB      0.2 MiB       import idb.fileformat
    38                             
    39                                 # buf = memview(buf)
    40     46.7 MiB      0.0 MiB       db = idb.fileformat.IDB(buf)
    41   1071.5 MiB   1024.8 MiB       db.vsParse(buf)
    42   1071.5 MiB      0.0 MiB       return db
```

And it doesn’t seem to be released, at out of scope. Even the garbage collector.

**Comment by imbillow** [comment]
image

What's this...

**Comment by williballenthin** [comment]
image

doesn't even give any details why it failed. weird!

File: tests/test_idaapi.py [hunk 1]
```diff
@@ -795,21 +787,22 @@ def test_exports(kernel32_idb, version, bitness, expected):
...
-    assert api.idc.GetType(0x6899AE01) == None
+    assert api.idc.GetType(0x6899AE01) is None
 
 
 @kern32_test()
 def test_inf_structure(kernel32_idb, version, bitness, expected):
     api = idb.IDAPython(kernel32_idb)
     inf_structure = api.idaapi.get_inf_structure()
-    assert inf_structure.procname == 'metapc'
+    assert inf_structure.procName == 'metapc'
```
Comment (williballenthin) [inline tests/test_idaapi.py:805]: good catch!

**Comment by williballenthin** [comment]
I'm not a huge fan of reformatting the source for the sake of it, but if it was automatically done, and we can configure GH actions to reformat using a consistent style on every commit, then I think this is worthwhile.

**Comment by XVilka** [comment]
I think the failed check might be just a bug of GitHub/CI itself. Could be safely ignored.

**Comment by imbillow** [comment]
I add `black` to format code, and add git hooks to auto format and run `QA checks` before every commit.

But it need config the `hooksPath` use:

```
git config core.hooksPath ./.githooks
```

**Comment by imbillow** [comment]
> 
> 
> I think the failed check might be just a bug of GitHub/CI itself. Could be safely ignored.
> I am not very comfortable with disabling Python 2.7 on both Linux and Windows though, it represents a lot of users.

I agree with you, but after going through some profilers, I still don't know how to reduce the memory overhead of pytest.

It needs to consume 13G RAM now In Python 2.7 on Linux (and only needs about 5.5G RAM  in Python 3.8), 
but GitHub Action runner only has 7G RAM available, and it seems that only more RAM can be used through Self-hosted runners RAM.

**Comment by williballenthin** [comment]
does it take 13GB to process a single .idb, or is that all the memory used by different test cases in the same process? if the latter, could we split up the test runs into multiple processes?

**Comment by imbillow** [comment]
> does it take 13GB to process a single .idb, or is that all the memory used by different test cases in the same process? if the latter, could we split up the test runs into multiple processes?

The latter. Yeah, we can split up the test runs into multiple processes. I just think it's not very elegant, but it does solve the problem.

**Comment by XVilka** [comment]
Looks like these jobs stuck forever. Not sure if it's related with GitHub outage or the python-idb own problem.

**Comment by imbillow** [comment]
I re-run the CI on my folk repo, it just remains the python2.7 on ubuntu-latest failed.

image

I can't find any log about the **Run tests** pass, even while I find it in the log archive.

image

And the raw logs of **Python 2.7 at ubuntu-latest**:

```
2020-06-29T14:27:15.5386044Z ##[section]Starting: Request a runner to run this job
2020-06-29T14:27:16.1287393Z Can't find any online and idle self-hosted runner in current repository that matches the required labels: 'ubuntu-latest'
2020-06-29T14:27:16.1287430Z Can't find any online and idle self-hosted runner in current repository's account/organization that matches the required labels: 'ubuntu-latest'
2020-06-29T14:27:16.1287453Z Found online and idle hosted runner in current repository's account/organization that matches the required labels: 'ubuntu-latest'
2020-06-29T14:27:16.3268744Z ##[section]Finishing: Request a runner to run this job

```

So I think it can be considered as a bug of GitHub Action.

**Comment by XVilka** [comment]
So I think it's in a good shape to be merged.

**Pull request merged**